pkg/apisix/cluster.go

```diff
@@ -290,6 +310,51 @@ func (c *cluster) GlobalRule() GlobalRule {
 	return c.globalRules
 }
 
+// HealthCheck implements Cluster.HealthCheck method.
+func (c *cluster) HealthCheck(ctx context.Context) (err error) {
+	if c.cacheSyncErr != nil {
```

Comment:
Actually, I think health check should be independent with the first time cache sync.

Reply:
You mean health check should wait first time cache sync finished ?

Reply:
```
func (c *cluster) HealthCheck(ctx context.Context) (err error) {
	if c.cacheSyncErr != nil {
		err = c.cacheSyncErr
		return
	}
        // _cacheSyncing means the first time cache sync
	if atomic.LoadInt32(&c.cacheState) == _cacheSyncing { 
		return
	}
...
```

Reply:
Yep.

Reply:
> You mean health check should wait first time cache sync finished ?

The code above already implement it.

- The cluster sync have two state `syncing` `synced` only.
- When state is `synced` and has an error of health check. Just give up leader and re-create cluster to syncing.